cpp/tests/structure/graph_mask_test.cpp

```diff
@@ -0,0 +1,262 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governin_from_mtxg permissions and
+ * limitations under the License.
+ */
+
+#include <utilities/base_fixture.hpp>
+#include <utilities/test_graphs.hpp>
+#include <utilities/test_utilities.hpp>
+
+#include <cugraph/graph.hpp>
+#include <cugraph/graph_mask.hpp>
+#include <cugraph/graph_view.hpp>
+
+#include <raft/cudart_utils.h>
+#include <raft/handle.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <limits>
+#include <numeric>
+#include <tuple>
+#include <vector>
+
+template <bool store_transposed, typename vertex_t, typename edge_t, typename weight_t>
+std::tuple<std::vector<edge_t>, std::vector<vertex_t>, std::optional<std::vector<weight_t>>>
+graph_reference(vertex_t const* p_src_vertices,
+                vertex_t const* p_dst_vertices,
+                std::optional<weight_t const*> p_edge_weights,
+                vertex_t number_of_vertices,
+                edge_t number_of_edges)
+{
+  std::vector<edge_t> offsets(number_of_vertices + 1, edge_t{0});
+  std::vector<vertex_t> indices(number_of_edges, vertex_t{0});
+  auto weights = p_edge_weights
+                   ? std::make_optional<std::vector<weight_t>>(number_of_edges, weight_t{0.0})
+                   : std::nullopt;
+
+  for (edge_t i = 0; i < number_of_edges; ++i) {
+    auto major = store_transposed ? p_dst_vertices[i] : p_src_vertices[i];
+    offsets[1 + major]++;
+  }
+  std::partial_sum(offsets.begin() + 1, offsets.end(), offsets.begin() + 1);
+
+  for (edge_t i = 0; i < number_of_edges; ++i) {
+    auto major           = store_transposed ? p_dst_vertices[i] : p_src_vertices[i];
+    auto minor           = store_transposed ? p_src_vertices[i] : p_dst_vertices[i];
+    auto start           = offsets[major];
+    auto degree          = offsets[major + 1] - start;
+    auto idx             = indices[start + degree - 1]++;
+    indices[start + idx] = minor;
+    if (p_edge_weights) { (*weights)[start + idx] = (*p_edge_weights)[i]; }
+  }
+
+  return std::make_tuple(std::move(offsets), std::move(indices), std::move(weights));
+}
+
+struct Graph_Usecase {
+  bool test_weighted{false};
+  bool multigraph{false};
+  bool check_correctness{true};
+};
+
+template <typename input_usecase_t>
+class Tests_GraphMask
+  : public ::testing::TestWithParam<std::tuple<Graph_Usecase, input_usecase_t>> {
+ public:
+  Tests_GraphMask() {}
+  static void SetupTestCase() {}
```

Comment:
Set"U"pTestCase, this is a typo in the current cuGraph implementation, should be fixed after PR 2439 gets merged for the rest part.